src/chain/mod.rs

```diff
@@ -538,6 +679,118 @@ impl ChainSource {
 
 				res
 			},
+			Self::Electrum {
+				electrum_runtime_client,
+				onchain_wallet,
+				onchain_wallet_sync_status,
+				kv_store,
+				logger,
+				node_metrics,
+				..
+			} => {
+				let electrum_client: Arc<ElectrumRuntimeClient> =
+					if let Some(client) = electrum_runtime_client.read().unwrap().as_ref() {
+						Arc::clone(client)
+					} else {
+						debug_assert!(
+							false,
+							"We should have started the chain source before syncing the onchain wallet"
+						);
+						return Err(Error::FeerateEstimationUpdateFailed);
+					};
+				let receiver_res = {
+					let mut status_lock = onchain_wallet_sync_status.lock().unwrap();
+					status_lock.register_or_subscribe_pending_sync()
+				};
+				if let Some(mut sync_receiver) = receiver_res {
+					log_info!(logger, "Sync in progress, skipping.");
+					return sync_receiver.recv().await.map_err(|e| {
+						debug_assert!(false, "Failed to receive wallet sync result: {:?}", e);
+						log_error!(logger, "Failed to receive wallet sync result: {:?}", e);
+						Error::WalletOperationFailed
+					})?;
+				}
+
+				// If this is our first sync, do a full scan with the configured gap limit.
+				// Otherwise just do an incremental sync.
+				let incremental_sync =
+					node_metrics.read().unwrap().latest_onchain_wallet_sync_timestamp.is_some();
+
+				fn apply_wallet_update(
+					update_res: Result<BdkUpdate, Error>, incremental_sync: bool, now: Instant,
```

Comment:
Can this work as a closure too with fewer params? Some of these parameters don't seem to be used further down in the function, so maybe there is no problem with move/borrow. Not sure how it works exactly with rust.